src/Types.ts

```diff
@@ -165,11 +171,17 @@ export interface ConditionArgsBase {
   type: ConditionType
 }
 
+export interface TokenWithDecimalsArgs extends TokenArgs {
+  decimals: number;
```

Comment:
@phanpak decimals is not a number. number is floating point like 12.34. decimals should be `BigIntish`